### PR TITLE
IOSXR - Fix for issue#492 -ShowBgpInstanceAllAll-- dotted ASN not captured

### PR DESCRIPTION
**Pull request by amrit-ma-siu**
## Description
Fix RegEx to capture dotted Notation ASN

## Motivation and Context
Local-AS / BGP routes with dotted Notation ASN was not captured by the current parser

## Impact (If any)
N/A

## Screenshots:
genieparse_iosxr_show_bgp_test20210808

genieparse_iosxr_show_bgp_test20210808-all

## Checklist:

- [x] I have updated the changelog.
- [ ] I have updated the documentation (If applicable).
- [x ] I have added tests to cover my changes (If applicable).
- [x] All new and existing tests passed.
- [ ] All new code passed compilation.

### CODE REVIEW

**Review by GerriorL — APPROVED** [review]
Nice work! For future reference you don't need to add both folder tests and unittests, since we're migrating to folder tests.

File: changelog/undistributed/changelog_iosxr_show_bgp_instance_all_all_all_20210808.rst [hunk 1]
```diff
@@ -0,0 +1,11 @@
+--------------------------------------------------------------------------------
+                            Fix
+--------------------------------------------------------------------------------
+* IOSXR
+    * Modified ShowBgpInstanceAllAllSchema:
+        * Modified key 'local_as' to Or(int, str). This captures dotted Notation ASN which is string.
+    * Modified ShowBgpInstanceAllAll:
+        * Modified RegEx <p6>,<p16_2>,<p16>,<m3> under <p16>, <p17> to capture dotted Notation ASN in BGP
+    * Modified UT test_show_bgp.py to test output with dotted ASN
+    * Added UT folder for ShowBgpInstanceAllAll for output with dotted ASN
```
Comment (domachad) [inline changelog/undistributed/changelog_iosxr_show_bgp_instance_all_all_all_20210808.rst:10]: No need to mention unit tests in the changelog

File: src/genie/libs/parser/iosxr/show_bgp.py [hunk 2]
```diff
@@ -5062,7 +5063,7 @@ def cli(self, vrf_type='all', address_family='', instance='all', vrf='all', outp
 
         # 2219             0 200 33299 51178 47751 {27016} e
         p16_2 = re.compile(r'^\s*(?P<metric>[0-9]+) +(?P<weight>[0-9]+)'
-                           r' +(?P<path>[0-9\{\}\s]+) '
+                           r' +(?P<path>[0-9\.\{\}\s]+) '
```
Comment (domachad) [inline src/genie/libs/parser/iosxr/show_bgp.py:5066]: Can you please add a comment with an example that supports this change?

File: src/genie/libs/parser/iosxr/show_bgp.py [hunk 3]
```diff
@@ -5078,12 +5079,13 @@ def cli(self, vrf_type='all', address_family='', instance='all', vrf='all', outp
...
         # *>i192.168.111.0/24       10.189.99.98                                                    0       0 i
+        # *> 10.7.7.7/32        10.10.10.107             0             0 65107.65107 ?
         p16 = re.compile(r'^(?P<status_codes>(i|s|x|S|d|h|\*|\>|\s)+)'
                          r' *(?P<prefix>(?P<ip>[0-9\.\:\[\]]+)/(?P<mask>\d+))?'
-                         r' +(?P<next_hop>\S+) +(?P<number>[\d\s\{\}]+)'
+                         r' +(?P<next_hop>\S+) +(?P<number>[\d\.\s\{\}]+)'
                          r'(?: *(?P<origin_codes>(i|e|\?)))?$')
 
-        p17 = re.compile(r'(?P<path>[\d\s]+)'
+        p17 = re.compile(r'(?P<path>[\d\.\s]+)'
```
Comment (domachad) [inline src/genie/libs/parser/iosxr/show_bgp.py:5088]: Please add a comment with an example

**Review by domachad — COMMENTED** [review]
Great job! Thank you for your contribution! There are just a few details that need to be addressed before we can merge these changes.  
Please read the comments below.
Also, as Liam said, you don't need to add unit tests to `test_show_bgp.py` as we are using folder-based tests now.

**Comment by amrit-ma-siu** [comment]
> Nice work! For future reference you don't need to add both folder tests and unittests. Since we're migrating to folder tests we only need those.

Noted, Thanks.

File: changelog/undistributed/changelog_iosxr_show_bgp_instance_all_all_all_20210808.rst (hunk 1 above)
Comment (amrit-ma-siu) [reply changelog/undistributed/changelog_iosxr_show_bgp_instance_all_all_all_20210808.rst:10]: removed from changelog

File: src/genie/libs/parser/iosxr/show_bgp.py (hunk 2 above)
Comment (amrit-ma-siu) [reply src/genie/libs/parser/iosxr/show_bgp.py:5066]: I have added possible example, as the change was to capture dotted ASN.

File: src/genie/libs/parser/iosxr/show_bgp.py (hunk 3 above)
Comment (amrit-ma-siu) [reply src/genie/libs/parser/iosxr/show_bgp.py:5090]: I modified all RegEx used for capturing ASN in the path. And I am not able to replicate some of the output. But I put the possible example.

**Review by GerriorL — CHANGES_REQUESTED** [review]
Looks like some of the changes made are causing errors in the unittest based testing. Since it's passing the folder tests you can update the unittest to match, or remove it.

**Review by GerriorL — APPROVED** [review]

**Comment by GerriorL** [comment]
Usually when the test_fuzzy_regex_search_command.py unittest is failing it means that the json that finds the parsers isn't up to date. This can be fixed by running "make json" from the genieparser/ directory.

**Review by domachad — APPROVED** [review]

**Pull request merged**